examples/defaulter-gen/generators/defaulter.go

```diff
@@ -731,6 +803,26 @@ func (n *callNode) writeCalls(varName string, isVarPointer bool, sw *generator.S
 	}
 }
 
+func (n *callNode) writeDefaulter(varName string, isVarPointer bool, sw *generator.SnippetWriter) {
+	accessor := varName
+	if !isVarPointer {
+		accessor = "&" + accessor
+	}
+	if n.defaultValue != "" {
+		sw.Do("if reflect.ValueOf($.var$).IsNil() {\n", generator.Args{
```

Comment:
that's not great, IsNil only works for specific types (pointer and pointer-like types)

Reply:
What about [`IsZero`](https://golang.org/pkg/reflect/#Value.IsZero) instead?

Reply:
Thanks for the catch, updated.